proposals/lip-0053.md

```diff
@@ -54,23 +54,22 @@ The `inboxUpdate` contains the information relative to the messages to be includ
+##### messageWitnessHashes
 
-In the sending chain, all cross-chain messages are added to a regular Merkle tree attested by the outbox root. The `messageWitness` contains two values relative to that Merkle tree:
-the `partnerChainOutboxSize`, giving the total messages sent by the partner chain, and the `siblingHashes`, which is part of a regular Merkle proof.
+In the sending chain, all cross-chain messages are added to a regular Merkle tree attested by the outbox root. The `messageWitnessHashes` are the sibling hashes necessary for the Merkle proof.
 
 This Merkle proof is required if not all messages necessary to recompute the outbox root were given in `crossChainMessages`. If all messages required to compute the outbox root are included in `crossChainMessages` then `siblingHashes` can be left empty.
 
 ##### outboxRootWitness
 
-The outbox root of the sending chain is part of a sparse Merkle tree attested by the state root (provided in the certificate). The `outboxRootWitness` property is an inclusion proof into the state root of the outbox root computed from the receiving chain inbox, `crossChainMessages` and `messageWitness`. If the cross-chain update transaction contains an empty certificate, this property can also be left empty, as the required root was already attested in a previous cross-chain update.
+The outbox root of the sending chain is part of a sparse Merkle tree attested by the state root (provided in the certificate). The `outboxRootWitness` property is an inclusion proof into the state root of the outbox root computed from the receiving chain inbox, `crossChainMessages` and `messageWitnessHashes`. If the cross-chain update transaction contains an empty certificate, this property can also be left empty, as the required root was already attested in a previous cross-chain update.
 
 ### Posting Cross-chain Updates on Mainchain and on Sidechains
 
 As the roles of both mainchain and sidechain are quite different, so are the transactions used to post cross-chain updates. The most notable differences are:
 
 * On the mainchain, all CCMs included in the `inboxUpdate` must have their `sendingChainID` equal to the chainID sending the cross-chain update.
-* On sidechains, all CCMs included in the `inboxUpdate` must have their `receivingChainID` equal to the chainID of the sidechain receiving the cross-chain update.
+* On sidechains, all CCMs included in the `inboxUpdate` must have their `receivingChainID` equal to the chainID of the sidechain receiving the cross-chain update (if has the default "ok" status).
 * As the mainchain accepts CCMs with `receivingChainID` corresponding to another sidechain, the cross-chain update processing on the mainchain is responsible for the inclusion of those CCMs in the corresponding outbox.
 
 These points guarantee that the CCMs are always forwarded to the correct chains and that the receiving chain can be confident that the chain specified in `ccm.sendingChainID` was truly the chain issuing the CCM.
```

Comment:
The rationale here is a bit incomplete in my opinion. If a blockchain is following the interoperability protocol, all messages are authenticated by the outbox root and the above properties are guaranteed. The main issue that we prevent here is that a malicious sidechain not following the interoperability protocol can spoof messages.

Reply:
But I think we make this point in the text: 'the receiving chain can be confident that the chain specified in `ccm.sendingChainID` was truly the chain issuing the CCM.'

Reply:
I guess then this is mentioned, can be resolved.